plugin/core/edit.py

```diff
@@ -26,38 +22,32 @@ def parse_workspace_edit(workspace_edit: WorkspaceEdit) -> Dict[str, List[TextEd
             text_document = document_change["textDocument"]
             uri = text_document['uri']
             version = text_document.get('version')
-            text_edit = list(parse_text_edit(change, version) for change in document_change.get('edits'))
-            changes.setdefault(uri, []).extend(text_edit)
+            edits = document_change.get('edits')
+            changes.setdefault(uri, ([], version))[0].extend(edits)
     else:
         raw_changes = workspace_edit.get('changes')
         if isinstance(raw_changes, dict):
-            for uri, uri_changes in raw_changes.items():
-                changes[uri] = list(parse_text_edit(change) for change in uri_changes)
+            for uri, edits in raw_changes.items():
+                changes[uri] = (edits, None)
     return changes
 
 
 def parse_range(range: Position) -> Tuple[int, int]:
     return range['line'], min(UINT_MAX, range['character'])
 
 
-def parse_text_edit(text_edit: TextEdit, version: Optional[int] = None) -> TextEditTuple:
-    return (
-        parse_range(text_edit['range']['start']),
-        parse_range(text_edit['range']['end']),
-        # Strip away carriage returns -- SublimeText takes care of that.
-        text_edit.get('newText', '').replace("\r", ""),
-        version
+def apply_text_edits(
+    view: sublime.View,
+    edits: Optional[List[TextEdit]],
+    *,
+    process_placeholders: Optional[bool] = False,
+    required_view_version: Optional[int] = None
+) -> None:
+    view.run_command(
+        'lsp_apply_document_edit',
+        {
+            'changes': edits or [],
+            'process_placeholders': process_placeholders,
+            'required_view_version': required_view_version,
+        }
     )
```

Comment:
I would do it like this:
```suggestion
    if edits:
        view.run_command(
            'lsp_apply_document_edit',
            {
                'changes': edits,
                'process_placeholders': process_placeholders,
                'required_view_version': required_view_version,
            }
        )
```

Reply:
addressed with early exit to avoid extra indent

Reply:
You can remove the ` or []` now